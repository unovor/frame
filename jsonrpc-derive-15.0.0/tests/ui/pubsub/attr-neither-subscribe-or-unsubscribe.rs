#[macro_use]
extern crate jsonrpc_derive;
extern crate jsonrpc_core;
extern crate jsonrpc_pubsub;

#[rpc]
pub trait Rpc {
	type Metadata;

	/// Hello subscription
	#[pubsub(subscription = "hello", name = "hello_subscribe", alias("hello_sub"))]
	fn subscribe(&self, _: Self::Metadata, _: typed::Subscriber<String>, _: u64);

	/// Unsubscribe from hello subscription.
	#[pubsub(subscription = "hello", unsubscribe, name = "hello_unsubscribe")]
	fn unsubscribe(&self, _: Option<Self::Metadata>, _: SubscriptionId) -> Result<bool>;
}

fn main() {}
