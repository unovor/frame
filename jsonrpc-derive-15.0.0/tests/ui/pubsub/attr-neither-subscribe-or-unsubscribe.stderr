error: pubsub attribute not annotated with either subscribe or unsubscribe
  --> $DIR/attr-neither-subscribe-or-unsubscribe.rs:10:2
   |
10 |       /// Hello subscription
   |  _____^
11 | |     #[pubsub(subscription = "hello", name = "hello_subscribe", alias("hello_sub"))]
12 | |     fn subscribe(&self, _: Self::Metadata, _: typed::Subscriber<String>, _: u64);
   | |_________________________________________________________________________________^

error: aborting due to previous error

